modules/nf-core/ctree/templates/main_script.R

```diff
@@ -0,0 +1,198 @@
+
+    CCF_table = add_dummy_driver(CCF_table, variant_colname="variantID", is_driver_colname="is.driver")
+
+    if (nrow(drivers_table)==0) {
+        drivers_table = CCF_table %>%
+        dplyr::filter(is.driver) %>%
+        dplyr::select(-nMuts) %>%
+        dplyr::mutate(patientID=patient)
+    }
+
+    ctree_init = list("CCF_table"=CCF_table,
+                      "drivers_table"=drivers_table,
+                      "samples"=samples,
+                      "patient"=patient)
+    return(ctree_init)
+}
+
+if ( grepl(".rds\$", tolower("$ctree_input")) ) {
+    best_fit = readRDS("$ctree_input")
+    do_fit = TRUE
+
+    if (!"data" %in% names(best_fit)) { best_fit[["data"]] =  data.frame(row.names=1:best_fit[["N"]])}
+
+    ## viber
+    if (class(best_fit) == "vb_bmm") {
+        fn_name = VIBER::get_clone_trees
+        subclonal_tool = "VIBER"
+
+    # If only one sample inference ctree is not working with VIBER
+    if (ncol(best_fit[["x"]]) <= 2) {
+        cli::cli_alert_warning("Object of class {class(best_fit)} with only one sample is not supported by the function `get_clone_trees`!")
+        do_fit = FALSE
+    }
+
+    best_fit[["data"]] = add_dummy_driver(best_fit[["data"]], variant_colname="gene", is_driver_colname="driver")
+    }
+
+    ## mobster
+    if (class(best_fit) == "dbpmm") {
+        fn_name = mobster::get_clone_trees
+        subclonal_tool = "mobster"
+        sample_id = unique(best_fit[["data"]][["sample_id"]])
+        outdir = paste0(sample_id, "/")
+        best_fit[["data"]] = add_dummy_driver(best_fit[["data"]], variant_colname="driver_label", is_driver_colname="is_driver")
+    }
+
+    if (class(best_fit) %in% c("vb_bmm", "dbpmm") & do_fit) {
+        # VIBER or MOBSTER object
+        trees = fn_name(x = best_fit)
+    } else {
+        cli::cli_alert_warning("Object of class {class(best_fit)} not supported.")
+        do_fit = FALSE
+    }
+
+} else {
+    do_fit = TRUE
+    subclonal_tool = "pyclonevi"
+    input_table = read.csv("$ctree_input", sep="\t")
+    data_ctree = initialize_ctree_obj_pyclone(input_table)
+    trees = ctrees(CCF_clusters = data_ctree[["CCF_table"]],
+                   drivers = data_ctree[["drivers_table"]],
+                   samples = data_ctree[["samples"]],
+                   patient = data_ctree[["patient"]])
+}
+
+if (do_fit & !is.null(trees)) {
+    ctree_output = paste0("ctree_", subclonal_tool)
+    if (outdir != "" & !dir.exists(outdir)) dir.create(outdir, recursive=T)
+
+    # plot the best tree
+    top_phylo = plot(trees[[1]])
+
+    # save rds and plots
+    saveRDS(object=trees, file=paste0(outdir, opt[["prefix"]], "_", ctree_output, ".rds"))
+    saveRDS(object=top_phylo, file=paste0(outdir, opt[["prefix"]], "_", ctree_output, "_plots.rds"))
+
+    # save report plot
+    phylos = ggplot2::ggplot()
+    if (length(trees) > 1) phylos = lapply(trees[2:min(length(trees), 3)], plot)
+    phylos = ggpubr::ggarrange(plotlist=phylos)
```

Comment:
ggpubr doesn't appear in your environment.yml file. Is it a dependency of one of the other packages?

Reply:
I would say yes, but I'm not sure. @SPPearce  - can you clarify if it should be in environment.yml?

Reply:
If it is being installed by conda through another package you don't need to add it to the environment.yml separately.